sentry_sdk/integrations/celery.py

```diff
@@ -140,13 +140,21 @@ def apply_async(*args, **kwargs):
         # type: (*Any, **Any) -> Any
         hub = Hub.current
         integration = hub.get_integration(CeleryIntegration)
-        if integration is not None and integration.propagate_traces:
+
+        # Note: kwargs can contain headers=None, so no setdefault!
+        # Unsure which backend though.
+        kwarg_headers = kwargs.get("headers") or {}
+        propagate_traces = kwarg_headers.pop("sentry-propagate-traces", None)
+        if propagate_traces is None and integration is not None:
```

Comment:
If you go with my previous suggestion:
```suggestion
        if propagate_traces is None:
```

Reply:
done